alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportExecuteWorkflowAction.kt

```diff
@@ -0,0 +1,121 @@
+                    try {
+                        val workflowRunResult =
+                            WorkflowRunnerService.runJob(workflow, periodStart, periodEnd, execWorkflowRequest.dryrun)
+                        withContext(Dispatchers.IO) {
+                            actionListener.onResponse(
+                                ExecuteWorkflowResponse(
+                                    workflowRunResult
+                                )
+                            )
+                        }
+                    } catch (e: Exception) {
+                        log.error("Unexpected error running workflow", e)
+                        withContext(Dispatchers.IO) {
+                            actionListener.onFailure(AlertingException.wrap(e))
+                        }
+                    }
+                }
+            }
+
+            if (execWorkflowRequest.workflowId != null) {
+                val getRequest = GetRequest(ScheduledJob.SCHEDULED_JOBS_INDEX).id(execWorkflowRequest.workflowId)
+                client.get(
+                    getRequest,
+                    object : ActionListener<GetResponse> {
+                        override fun onResponse(response: GetResponse) {
+                            if (!response.isExists) {
+                                actionListener.onFailure(
+                                    AlertingException.wrap(
+                                        OpenSearchStatusException(
+                                            "Can't find workflow with id: ${response.id}",
+                                            RestStatus.NOT_FOUND
+                                        )
+                                    )
+                                )
+                                return
+                            }
+                            if (!response.isSourceEmpty) {
+                                XContentHelper.createParser(
+                                    xContentRegistry, LoggingDeprecationHandler.INSTANCE,
+                                    response.sourceAsBytesRef, XContentType.JSON
+                                ).use { xcp ->
+                                    val workflow = ScheduledJob.parse(xcp, response.id, response.version) as Workflow
+                                    executeWorkflow(workflow)
+                                }
+                            }
+                        }
+
+                        override fun onFailure(t: Exception) {
+                            actionListener.onFailure(AlertingException.wrap(t))
+                        }
+                    }
+                )
+            } else {
+                val workflow = when (user?.name.isNullOrEmpty()) {
+                    true -> execWorkflowRequest.workflow as Workflow
+                    false -> (execWorkflowRequest.workflow as Workflow).copy(user = user)
+                }
+                executeWorkflow(workflow)
```

Comment:
Ah I see. Very very good point.

Reply:
I think we don't need to do the same thing when executing the workflow.
why?  Because workflow in itself will contain the list of delegate monitor ids (which are needed to exist - if not, they won't be executed). meaning that, if some of the monitorIds are document level monitors (which let's say exist), then it would mean that those document level monitors already have docQuery index initialized
The workflows only work with pre-created monitors